src/Components/VotingPage/VotingPage.jsx

```diff
@@ -77,44 +89,48 @@ export default function VotingPage() {
   };
 
   const downloadDesigns = async () => {
-    const imageURLRef = doc(db, "images", votingPageID);
+    const imageURLRef = doc(db, "VotingPages", votingPageUrl);
     const docSnap = await getDoc(imageURLRef);
     setDesign1DownloadUrl(docSnap.data().image1DownloadUrl);
     setDesign2DownloadUrl(docSnap.data().image2DownloadUrl);
   };
 
   const uploadImages = async () => {
-    const docRef = await addDoc(collection(db, "images"), {
+    const generatedPageUrl = uuidv4();
+    setVotingPageUrl(generatedPageUrl);
+    console.log("generated page url : \n" + generatedPageUrl);
+    await setDoc(doc(db, "VotingPages", generatedPageUrl), {
       timestamp: serverTimestamp(),
-      name: "owner1",
-      design1Votes: 2,
-      design2Votes: 3,
+      userID: "12345",
+      votingPageUrl: generatedPageUrl,
+      design1Votes: 0,
+      design2Votes: 0,
     });
-
+    const docRef = doc(db, "VotingPages", generatedPageUrl);
     const docSnap = await getDoc(docRef);
     setDesign1VoteNumbers(docSnap.data().design1Votes);
     setDesign2VoteNumbers(docSnap.data().design2Votes);
     await Promise.all(
       dropedImages1.map((image) => {
```

Comment:
Line 114:33:  Array.prototype.map() expects a return value from arrow function

Reply:
The warning is gone, you can test it now